apps/emqx_exhook/src/emqx_exhook_mgr.erl

```diff
@@ -181,10 +181,14 @@ pre_config_update(_, {enable, Name, Enable}, OldConf) ->
     of
         not_found -> throw(not_found);
         NewConf -> {ok, lists:map(fun maybe_write_certs/1, NewConf)}
-    end.
+    end;
+pre_config_update(_, NewConf, _OldConf) when NewConf =:= #{} ->
+    {ok, NewConf#{<<"servers">> => []}};
+pre_config_update(_, NewConf = #{<<"servers">> := Servers}, _OldConf) ->
+    {ok, NewConf#{<<"servers">> => lists:map(fun maybe_write_certs/1, Servers)}}.
```

Comment:
it feels safer to actually use a new update request label for root level reloading.
e.g. `root_reload`

so the handlers can implement more explicit matching clauses.

Reply:
I don't think it's necessary to add this label. It will confuse the callers why emqx:update([rootKey,subKey],xxxx) works, but emqx:update([rootKey],yyyy) doesn't work. I personally think that providing a simple and unified external API is better, instead of adding a reload_rootkey label to each rootkey.